pennylane/templates/embeddings/amplitude.py

```diff
@@ -32,7 +30,7 @@
 
 
 @template
-def AmplitudeEmbedding(features, wires, pad=None, normalize=False):
+def AmplitudeEmbedding(features, wires, pad=None, normalize=True):
```

Comment:
In most cases `math.isclose` would be stricter and users would have to pass `normalize=True`, else an error as follows arises:
```python
ValueError: 'features' must be a vector of length 1.0; got length 0.9999999403953552.Use 'normalization=True' to automatically normalize.
```

@mariaschuld it was a major question mark if this would not be ideal. Please let me know if we'd **not** like to change the default here :thinking:

Reply:
:+1: I would like normalize by default

Reply:
This change feels a bit out of scope for this PR. I recall there was quite a big of discussion regarding the original default being `False`.

Reply:
It was added here because simply turning to `math.isclose` makes some tests fail due to tolerance issues. What were the reasons for keeping the default `False` for normalize? Could it affect performance?

An alternative that comes to mind would be raising tolerance from `1e-10` to `1e-6` both for the check in `default.qubit` and for `AmplitudeEmbedding`. To me, this seems like a change that would have more impact on the entire codebase (e.g. affects `default.qubit` even if not used with `AmplitudeEmbedding`).

The previously mentioned `ValueError` is raised for [torch integration tests](https://github.com/PennyLaneAI/pennylane/blob/d6ed14b45f7fa86651118c423d45bc4d74f38bd6/tests/interfaces/test_torch.py#L941):
```python
In [5]: a = torch.tensor([0, 1, 1, 0], requires_grad=False) / np.sqrt(2)        

In [6]: np.linalg.norm(a, ord=2)                                                
Out[6]: 0.99999994
```

Reply:
If I remember/read correctly normalisation works no matter if `features` is diffable or not, and I'd be happy to change the default here! In fact, one could drop the argument altogether and always normalise - the only use case I'd imagine is where the user wants control over the precision. 

As mentioned before, we should make the precision an argument to the template to give users more control of fixing bugs?